packages/backend/src/models/GroupsModel.ts

```diff
@@ -221,4 +223,54 @@ export class GroupsModel {
             organizationUuid: updatedGroup.organization_uuid,
         };
     }
+
+    async addUserToGroupsIfExist({
+        userUuid,
+        userId,
+        groups,
+        organizationUuid,
+    }: {
+        userUuid?: string;
+        userId?: number;
+        groups: string[];
+        organizationUuid: string;
+    }) {
+        const organization = await this.database('organizations')
+            .where('organization_uuid', organizationUuid)
+            .first('organization_id');
+        if (!organization) {
+            throw new NotExistsError('Cannot find organization');
+        }
+
+        const existingGroups = await this.database('groups')
+            .whereIn('name', groups)
+            .andWhere('organization_id', organization.organization_id)
+            .select('group_uuid', 'organization_id');
+
+        if (existingGroups.length === 0) {
+            return;
+        }
+
+        const userIdToInsert =
+            userId ??
+            (
+                await this.database('users')
+                    .where('user_uuid', userUuid)
+                    .first('user_id')
+            )?.user_id;
+        if (!userIdToInsert) {
+            throw new NotExistsError('Cannot find user');
+        }
+
+        const insertData = existingGroups.map((group) => ({
+            group_uuid: group.group_uuid,
+            user_id: userIdToInsert,
+            organization_id: group.organization_id,
```

Comment:
awesome. to be explicit it would be better to also use the `organization_id` from the organization not those attached to the groups returned from the select statement before

Reply:
sure! Sounds good :+1:

Reply:
https://github.com/lightdash/lightdash/pull/8508/commits/fa50b43f1df2e1686ef1c130d74154c573f37f74 and here it is!